Merge pull request #1 from dpastoor/master

nmehta2 opened the pull request:
let's try this again

nmehta2 committed:
Merge pull request #1 from dpastoor/master

let's try this again

dpastoor commented on lo_2010.cpp:
```diff
@@ -0,0 +1,51 @@
+wt  : 2.9   : Weight (kg)
+pma : 34.8  : post-menstral age (wk)
+scr : 0.9   : serum creatinine (mg/dL)
+Fsga :   ?  : small-for-gestational-age
```
Set default of 1

dpastoor commented on lo_2010.cpp:
```diff
@@ -0,0 +1,51 @@
+pma : 34.8  : post-menstral age (wk)
+scr : 0.9   : serum creatinine (mg/dL)
+Fsga :   ?  : small-for-gestational-age
+spiro :   ? : spironolactone (L/kg)
```
set default of 0

nmehta2 committed:
lo_2010

wrote the code!

questions (by section):
[PARAM]
- did not know values for Fsga and spiro (is this correct?)

[MAIN]
- did not know values for Fsga and spiro

[OMEGA], [SIGMA], [TABLE]
- have not made changes

[CAPTURE]
- is my code correct?

dpastoor commented on lo_2010.cpp:
```diff
@@ -0,0 +1,51 @@
+D_CENT = 1; 
+double NORM_WT = 70;
+double NORM_PMA = 30;
+double Fsga = ?;
```
this is how you define values inside your code - you have already set these as parameters so you don't need to re-define them

dpastoor commented:
so keep in mind distinguishing between parameters for input, and things you define in `[MAIN]`

the syntax of `double <value> = ...` is you initializing a value to use later on. For exaple the normalization value for wt, etc. If you are already passing that value in as a parameter it is already available so you don't want to define.

nmehta2 commented:
Why can't you just use the number 70 instead of creating a parameter in `[MAIN]` for `NORM_WT`? Is it easier to have a parameter which you can change if the norm_wt value changes (ex. different patient demographic)?

nmehta2 committed:
lo_2010 changes

updated [PARAM]
updated [MAIN]

dpastoor commented:
good question - you are close - the biggest reason is that it makes it much clearer to read, especially as the number of parameters expands. For example

`pow(WT/70, 0.75)` doesn't necessarily convey why 70 was chosen and also starts to become harder to read when in a larger formula

whereas by defining the values above with names, it becomes clearer to read the equations, as well as update/change accordingly.

dpastoor commented on lo_2010.cpp:
```diff
@@ -22,8 +22,6 @@ ncmt=1, trans=11
 double NORM_PMA = 30;
-double Fsga = ?;
-double spiro = ?;
 double CLi = 1.0 * pow((wt/NORM_WT), 0.75) * pow((pma/NORM_PMA), 3.16) * Fsga
```
ok so looking in the actual publication, this F term is misleading. I originally thought it was a 1/0, but that wouldn't makes sense as if the baby wasn't small for gestantioal age, CLi would be 0. Looking into the actual publication, we see that the full CL equation was actually

```
CL = 1.0 * (wt/70)^0.75*(PMA/30)^3.16*[0.83*SGA + 1.03*(1-SGA)]
```

so we should expand this equation to just have the parameter be called SGA (with a default of 0) and then use this equation.

nmehta2 commented:
ah okay, that makes sense! Thank you for explaining!

nmehta2 committed:
moved lo_2010 and made edits

moved lo_2010 to models folder
made edits n [PARAM] and [MAIN] CLi equation as requested

nmehta2 committed:
edited [main] of anderson_2007

removed finotrope and fventilation in [MAIN]
what should i do with those parameters in [PARAM]?

nmehta2 committed:
edits to anderson_2007

changed [PARAM]
renamed Fvent and Finot
assigned parameter values
defined parameters

[MAIN]
changed Fvent and Finot

changed [CAPTURE]
added Fvent and Finot
defined Fvent and Finot

dpastoor commented on models/andeson_2007.cpp:
```diff
@@ -23,8 +23,8 @@ ncmt=1, trans=11
 double NORM_PMA = 34.8;
-double CLi = 3.83*pow(wt/NORM_WT,0.75)*(pow(pma, 3.68)/(pow(pma, 3.68) + pow(33.3, 3.68)))*(516*pow(e, (0.00823*((PMA-40)/52-40)))/scr)/6)*Fventilation;
-double Vi = 39.4*(wt/NORM_WT)*Finotrope;
+double CLi = 3.83*pow(wt/NORM_WT,0.75)*(pow(pma, 3.68)/(pow(pma, 3.68) + pow(33.3, 3.68)))*(516*pow(e, (0.00823*((PMA-40)/52-40)))/scr)/6)*Fvent;
```
so these are both incorrect. Think of what you are setting as of now. You are saying multiply by 0, so both CL/V will be 0.

In reality, similar to the lo paper, these are likely placeholders for more complex equations like `(1.5*on_vent)`. The `Fvent` stands, essentially, for fraction corresponding to vent status. We need to actually go through the original anderson paper to find the actual values

dpastoor commented on models/andeson_2007.cpp:
```diff
@@ -23,8 +23,8 @@ ncmt=1, trans=11
+double Vi = 39.4*(wt/NORM_WT)*Finot;
 
 [OMEGA] @annotated
   ECL : 0.04 : Eta on CL
```
we will also need to pull out the values for the random effects

dpastoor reviewed (changes requested):
getting closer!

nmehta2 committed:
edits to anderson_2007

[PARAM]
changed Fvent and Finot values and definitions (found in Table 1a of
paper)

[OMEGA]
changed values of ECL and EV (found in Table 1a of paper)

[SIGMA]
changed values of PROP and ADD (found in Table 1a of paper)

[CAPTURE]
changed definitions of Fvent and Finot

dpastoor commented on models/lo_2010.cpp:
```diff
@@ -0,0 +1,49 @@
+double Vi = 0.572 * (1-0.344*spiro) * wt
+
+[OMEGA] @annotated
+  ECL : 0.04 : Eta on CL
```
@nmehta2 so close! lets just get the lo omega and sigma values and i think you got it!

nmehta2 replied:
i updated the omega values and the residual error model values! are they correct?

dpastoor commented on models/lo_2010.cpp:
```diff
@@ -0,0 +1,49 @@
+  
+  [TABLE]
+double CP = CENT/Vi;
+double DV = CP*(1+PROP);
```
check the residual error structure as well. This is only proportional as of now - is there also an additive component?

nmehta2 replied:
added the additive comp! did i write it correctly?

dpastoor reviewed (changes requested):
so close!

nmehta2 committed:
changes to lo_2010

changed eta values and residual error values

dpastoor commented on models/lo_2010.cpp:
```diff
@@ -26,12 +26,12 @@ double CLi = 1.0 * pow((wt/NORM_WT), 0.75) * pow((pma/NORM_PMA), 3.16) * (0.83*S
 [OMEGA] @annotated
-  ECL : 0.04 : Eta on CL
-  EV  : 0.01 : Eta on V
+  ECL : 20.5 : Eta on CL
```
what are all these values? they still need to be reported as variances, so you were correct in that 0.04 --> 20%CV. Remember the variance to %CV calculation can be approximated by sqrt(variance)*100, so sqrt(0.04)*100 = 20%
